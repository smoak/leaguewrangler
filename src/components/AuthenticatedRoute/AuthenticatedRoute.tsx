import React, { SFC } from 'react';
import { Route, RouteProps, Redirect } from 'react-router';

export interface MappedStateProps {
  readonly isAuthenticated: boolean;
}

export type AuthenticatedRouteProps = MappedStateProps & RouteProps;

const AuthenticatedRoute: SFC<AuthenticatedRouteProps> = ({
  component: Component,
  isAuthenticated,
  ...props
}: {
  component: React.ComponentType<RouteProps>;
  isAuthenticated: boolean;
}) => {
  return (
    <Route
      {...props}
      render={props => {
        return isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect
            to={{
              pathname: '/login',
            }}
          />
        );
      }}
    />
  );
};

export default AuthenticatedRoute;
