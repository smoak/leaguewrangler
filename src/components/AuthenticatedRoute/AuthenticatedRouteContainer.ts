import AuthenticatedRoute, { MappedStateProps } from './AuthenticatedRoute';
import { MapStateToProps, connect } from 'react-redux';
import { ApplicationState } from '../../data/state/types';
import { RouteProps } from 'react-router';
import { isAuthenticated } from '../../data/domains/session/selectors';

export const mapStateToProps: MapStateToProps<MappedStateProps, RouteProps, ApplicationState> = state => {
  return {
    isAuthenticated: isAuthenticated(state),
  };
};

export default connect(mapStateToProps)(AuthenticatedRoute);
