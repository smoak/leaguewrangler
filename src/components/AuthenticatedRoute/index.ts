export { default } from './AuthenticatedRouteContainer';
export * from './AuthenticatedRoute';
