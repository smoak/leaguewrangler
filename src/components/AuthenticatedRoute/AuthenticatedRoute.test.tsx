import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import AuthenticatedRoute from './AuthenticatedRoute';

describe('AuthenticatedRoute', () => {
  let component: ShallowWrapper;

  describe('when not authenticated', () => {
    beforeEach(() => {
      component = shallow(<AuthenticatedRoute isAuthenticated={false} component={() => <>hey</>} />);
    });

    it('redirects to the login page', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when authenticated', () => {
    beforeEach(() => {
      component = shallow(<AuthenticatedRoute isAuthenticated={true} component={() => <>hey</>} />);
    });

    it('renders the component provided to the route', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
