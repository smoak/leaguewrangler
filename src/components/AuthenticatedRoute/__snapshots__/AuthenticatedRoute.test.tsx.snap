// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticatedRoute when authenticated renders the component provided to the route 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`AuthenticatedRoute when not authenticated redirects to the login page 1`] = `
<Route
  render={[Function]}
/>
`;
